asteroid/masknn/convolutional.py

```diff
@@ -491,6 +506,65 @@ class DCUMaskNet(BaseDCUMaskNet):  # CHECK-JIT
 
     _architectures = DCUNET_ARCHITECTURES
 
+    def __init__(self, encoders, decoders, fix_length_mode=None, **kwargs):
+        self.fix_length_mode = fix_length_mode
+        self.encoders_stride_product = np.prod(
+            [enc_stride for _, _, _, enc_stride, _ in encoders], axis=0
+        )
+
+        # Avoid circual import
+        from .convolutional import DCUNetComplexDecoderBlock, DCUNetComplexEncoderBlock
+
+        super().__init__(
+            encoders=[DCUNetComplexEncoderBlock(*args) for args in encoders],
+            decoders=[DCUNetComplexDecoderBlock(*args) for args in decoders[:-1]],
+            output_layer=complex_nn.ComplexConvTranspose2d(*decoders[-1]),
+            **kwargs,
+        )
+
+    def fix_input_dims(self, x):
+        return _fix_dcu_input_dims(
+            self.fix_length_mode, x, torch.from_numpy(self.encoders_stride_product)
+        )
+
+    def fix_output_dims(self, out, x):
+        return _fix_dcu_output_dims(self.fix_length_mode, out, x)
+
+
+@script_if_tracing
+def _fix_dcu_input_dims(fix_length_mode: Optional[str], x, encoders_stride_product):
+    """Pad or trim `x` to a length compatible with DCUNet."""
+    freq_prod = int(encoders_stride_product[0])
+    time_prod = int(encoders_stride_product[1])
+    if (x.shape[1] - 1) % freq_prod:
+        raise TypeError(
+            f"Input shape must be [batch, freq + 1, time + 1] with freq divisible by "
+            f"{freq_prod}, got {x.shape} instead"
+        )
+    time_remainder = (x.shape[2] - 1) % time_prod
+    if time_remainder:
+        if fix_length_mode is None:
+            raise TypeError(
+                f"Input shape must be [batch, freq + 1, time + 1] with time divisible by "
+                f"{time_prod}, got {x.shape} instead. Set the 'fix_length_mode' argument "
+                f"in 'DCUNet' to 'pad' or 'trim' to fix shapes automatically."
+            )
+        elif fix_length_mode == "pad":
+            pad_shape = [0, time_prod - time_remainder]
+            x = nn.functional.pad(x, pad_shape, mode="constant")
+        elif fix_length_mode == "trim":
+            pad_shape = [0, -time_remainder]
+            x = nn.functional.pad(x, pad_shape, mode="constant")
+        else:
+            raise ValueError(f"Unknown fix_length mode '{fix_length_mode}'")
+    return x
+
+
+@script_if_tracing
+def _fix_dcu_output_dims(fix_length_mode: Optional[str], out, x):
+    """Fix shape of `out` to the original shape of `x`."""
+    return pad_x_to_y(out, x)
```

Comment:
Checking if negative padding works here (trimming and padding in the same function) and solves the 1.7 error